packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -994,10 +1020,13 @@ export class UserPool extends UserPoolBase {
    * Add a lambda trigger to a user pool operation
    * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html
    */
-  public addTrigger(operation: UserPoolOperation, fn: lambda.IFunction): void {
+  public addTrigger(operation: UserPoolOperation, fn: lambda.IFunction, lambdaVersion?: LambdaVersion): void {
     if (operation.operationName in this.triggers) {
       throw new Error(`A trigger for the operation ${operation.operationName} already exists.`);
     }
+    if (operation !== UserPoolOperation.PRE_TOKEN_GENERATION_CONFIG && lambdaVersion === LambdaVersion.V2_0) {
```

Comment:
Do we have to check if `lambdaVersion` exists as well?

Reply:
I believe the only scenario where I want to raise an error is when the lambdaVersion is set to V2 and the operation is PRE_TOKEN_GENERATION. 
Since lambdaVersion defaults to V1 when undefined, I think checking for that is unnecessary. What do you think?

Reply:
my only concern was the check would throw an error saying the lambdaVersion doesn't exist (if the user left it empty). Did a quick check myself and realized thats not the case so i think its fine to leave as is. Might be good to have a test for it tho (if it doesn't exist already)

Reply:
Sure! I've added unit test.
https://github.com/aws/aws-cdk/pull/28899/files#diff-2c630fbb5d50f4f6d047a873debf0ab809aae98b8e00d9ab8d86c8ee4f3cdaf2R511

Reply:
beautiful